server/src/main/java/org/elasticsearch/action/support/UnsafePlainActionFuture.java

```diff
@@ -25,12 +25,21 @@ public class UnsafePlainActionFuture<T> extends PlainActionFuture<T> {
     private final String unsafeExecutor;
     private final String unsafeExecutor2;
 
+    /**
+     * Allow the single executor passed to be used unsafely. This allows waiting for the future and completing the future on threads in
+     * the same executor, but only for the specific executor.
+     */
     public UnsafePlainActionFuture(String unsafeExecutor) {
```

Comment:
Likewise how about this?

```suggestion
    public UnsafePlainActionFuture(String blockingPermittedExecutor) {
```

Reply:
I think that makes it less clear in that it is the act of blocking and completing on the same future that is allowed. Hence I went for unsafe in the class name and piggy back on it here (rather than explaining that too deeply in class and variable names)